One codebase for android and j2se (java+awt)

k3b opened the pull request:
## Subprojects
- pixymeta-lib
  - platform independant code useable by android and by j2se
  - has no code to handle (preview-)Bitmaps that are platform dependant.
  - if you want to use IBitmap (an interface replacement for Bitmap) you either need pixymeta-android-lib or pixymeta-j2se-lib
- pixymeta-android-lib
  - android specific implementation to handle (preview-)Bitmaps
- pixymeta-j2se-lib
  - java/awt specific implementation to handle (preview-)Bitmaps
- pixymeta-j2se-demo
  - java commandline programm to demonstrate image meta functions
## Dependencies
- implementation of imageio drivers 
  - com.twelvemonkeys.imageio:imageio-tiff (version 3.2.1 working with pixymeta-j2se-demo)
  - com.github.jai-imageio:jai-imageio-core (version 1.3.1 has problems with tiff files)
## How to build
- use `./gradlew --gui` 
  - to start a build gui
- use `./gradlew pixymeta-j2se-demo:fatJar` 
  - to create `pixymeta-j2se-demo-all.jar` that contains all dependencies and testimages
  - execute `java -jar pixymeta-j2se-demo-all.jar` to run the demo

k3b committed:
removed intermediate build files from repository

k3b committed:
removed maven specific files from repository

k3b committed:
refactored project layout to become compatible with gradle/android studio

k3b committed:
Merge pull request #1 from dragon66/master

added changes from origin

k3b committed:
Merge branch 'master' into gradle-build

Conflicts:
	bin/pixy/image/ImageType$1.class
	bin/pixy/image/ImageType$2.class
	bin/pixy/image/ImageType$3.class
	bin/pixy/image/ImageType$4.class
	bin/pixy/image/ImageType$5.class
	bin/pixy/image/ImageType$6.class
	bin/pixy/image/ImageType$7.class
	bin/pixy/image/ImageType$8.class
	bin/pixy/image/ImageType$9.class
	bin/pixy/image/ImageType.class
	bin/pixy/image/bmp/BmpCompression.class
	bin/pixy/image/gif/ApplicationExtension.class
	bin/pixy/image/jpeg/COMBuilder.class
	bin/pixy/image/jpeg/COMReader.class
	bin/pixy/image/jpeg/Component.class
	bin/pixy/image/jpeg/DHTReader.class
	bin/pixy/image/jpeg/DQTReader.class
	bin/pixy/image/jpeg/HTable.class
	bin/pixy/image/jpeg/JPEGConsts.class
	bin/pixy/image/jpeg/Marker$Attribute.class
	bin/pixy/image/jpeg/Marker.class
	bin/pixy/image/jpeg/QTable.class
	bin/pixy/image/jpeg/SOFReader.class
	bin/pixy/image/jpeg/SOSReader.class
	bin/pixy/image/jpeg/Segment.class
	bin/pixy/image/jpeg/SegmentBuilder.class
	bin/pixy/image/jpeg/UnknownSegment.class
	bin/pixy/image/png/Chunk.class
	bin/pixy/image/png/ChunkBuilder.class
	bin/pixy/image/png/ChunkType$Attribute$1.class
	bin/pixy/image/png/ChunkType$Attribute$2.class
	bin/pixy/image/png/ChunkType$Attribute.class
	bin/pixy/image/png/ChunkType.class
	bin/pixy/image/png/ColorType.class
	bin/pixy/image/png/Filter.class
	bin/pixy/image/png/ICCPBuilder.class
	bin/pixy/image/png/IDATBuilder.class
	bin/pixy/image/png/IDATReader.class
	bin/pixy/image/png/IENDBuilder.class
	bin/pixy/image/png/IENDReader.class
	bin/pixy/image/png/IHDRBuilder.class
	bin/pixy/image/png/IHDRReader.class
	bin/pixy/image/png/PLTEBuilder.class
	bin/pixy/image/png/PLTEReader.class
	bin/pixy/image/png/PNGDescriptor.class
	bin/pixy/image/png/SRGBBuilder.class
	bin/pixy/image/png/SRGBReader.class
	bin/pixy/image/png/TIMEBuilder.class
	bin/pixy/image/png/TIMEReader.class
	bin/pixy/image/png/TRNSBuilder.class
	bin/pixy/image/png/TRNSReader.class
	bin/pixy/image/png/TextBuilder.class
	bin/pixy/image/png/TextReader.class
	bin/pixy/image/png/UnknownChunk.class
	bin/pixy/image/png/UnknownChunkBuilder.class
	bin/pixy/image/png/UnknownChunkReader.class
	bin/pixy/image/tiff/ASCIIField.class
	bin/pixy/image/tiff/AbstractByteField.class
	bin/pixy/image/tiff/AbstractLongField.class
	bin/pixy/image/tiff/AbstractRationalField.class
	bin/pixy/image/tiff/AbstractShortField.class
	bin/pixy/image/tiff/ByteField.class
	bin/pixy/image/tiff/DoubleField.class
	bin/pixy/image/tiff/FieldType.class
	bin/pixy/image/tiff/FloatField.class
	bin/pixy/image/tiff/IFD.class
	bin/pixy/image/tiff/IFDField.class
	bin/pixy/image/tiff/LongField.class
	bin/pixy/image/tiff/RationalField.class
	bin/pixy/image/tiff/SByteField.class
	bin/pixy/image/tiff/SLongField.class
	bin/pixy/image/tiff/SRationalField.class
	bin/pixy/image/tiff/SShortField.class
	bin/pixy/image/tiff/ShortField.class
	bin/pixy/image/tiff/TIFFImage.class
	bin/pixy/image/tiff/Tag.class
	bin/pixy/image/tiff/TiffField.class
	bin/pixy/image/tiff/TiffFieldEnum$Compression.class
	bin/pixy/image/tiff/TiffFieldEnum$PhotoMetric.class
	bin/pixy/image/tiff/TiffFieldEnum$PlanarConfiguration.class
	bin/pixy/image/tiff/TiffFieldEnum.class
	bin/pixy/image/tiff/TiffTag$1.class
	bin/pixy/image/tiff/TiffTag$10.class
	bin/pixy/image/tiff/TiffTag$11.class
	bin/pixy/image/tiff/TiffTag$12.class
	bin/pixy/image/tiff/TiffTag$13.class
	bin/pixy/image/tiff/TiffTag$14.class
	bin/pixy/image/tiff/TiffTag$15.class
	bin/pixy/image/tiff/TiffTag$16.class
	bin/pixy/image/tiff/TiffTag$17.class
	bin/pixy/image/tiff/TiffTag$18.class
	bin/pixy/image/tiff/TiffTag$19.class
	bin/pixy/image/tiff/TiffTag$2.class
	bin/pixy/image/tiff/TiffTag$20.class
	bin/pixy/image/tiff/TiffTag$21.class
	bin/pixy/image/tiff/TiffTag$22.class
	bin/pixy/image/tiff/TiffTag$23.class
	bin/pixy/image/tiff/TiffTag$24.class
	bin/pixy/image/tiff/TiffTag$25.class
	bin/pixy/image/tiff/TiffTag$26.class
	bin/pixy/image/tiff/TiffTag$27.class
	bin/pixy/image/tiff/TiffTag$28.class
	bin/pixy/image/tiff/TiffTag$29.class
	bin/pixy/image/tiff/TiffTag$3.class
	bin/pixy/image/tiff/TiffTag$30.class
	bin/pixy/image/tiff/TiffTag$31.class
	bin/pixy/image/tiff/TiffTag$32.class
	bin/pixy/image/tiff/TiffTag$33.class
	bin/pixy/image/tiff/TiffTag$34.class
	bin/pixy/image/tiff/TiffTag$35.class
	bin/pixy/image/tiff/TiffTag$36.class
	bin/pixy/image/tiff/TiffTag$37.class
	bin/pixy/image/tiff/TiffTag$38.class
	bin/pixy/image/tiff/TiffTag$39.class
	bin/pixy/image/tiff/TiffTag$4.class
	bin/pixy/image/tiff/TiffTag$40.class
	bin/pixy/image/tiff/TiffTag$41.class
	bin/pixy/image/tiff/TiffTag$42.class
	bin/pixy/image/tiff/TiffTag$43.class
	bin/pixy/image/tiff/TiffTag$44.class
	bin/pixy/image/tiff/TiffTag$45.class
	bin/pixy/image/tiff/TiffTag$46.class
	bin/pixy/image/tiff/TiffTag$47.class
	bin/pixy/image/tiff/TiffTag$48.class
	bin/pixy/image/tiff/TiffTag$49.class
	bin/pixy/image/tiff/TiffTag$5.class
	bin/pixy/image/tiff/TiffTag$50.class
	bin/pixy/image/tiff/TiffTag$51.class
	bin/pixy/image/tiff/TiffTag$52.class
	bin/pixy/image/tiff/TiffTag$53.class
	bin/pixy/image/tiff/TiffTag$54.class
	bin/pixy/image/tiff/TiffTag$55.class
	bin/pixy/image/tiff/TiffTag$56.class
	bin/pixy/image/tiff/TiffTag$57.class
	bin/pixy/image/tiff/TiffTag$58.class
	bin/pixy/image/tiff/TiffTag$59.class
	bin/pixy/image/tiff/TiffTag$6.class
	bin/pixy/image/tiff/TiffTag$60.class
	bin/pixy/image/tiff/TiffTag$61.class
	bin/pixy/image/tiff/TiffTag$62.class
	bin/pixy/image/tiff/TiffTag$63.class
	bin/pixy/image/tiff/TiffTag$64.class
	bin/pixy/image/tiff/TiffTag$65.class
	bin/pixy/image/tiff/TiffTag$66.class
	bin/pixy/image/tiff/TiffTag$67.class
	bin/pixy/image/tiff/TiffTag$68.class
	bin/pixy/image/tiff/TiffTag$69.class
	bin/pixy/image/tiff/TiffTag$7.class
	bin/pixy/image/tiff/TiffTag$70.class
	bin/pixy/image/tiff/TiffTag$71.class
	bin/pixy/image/tiff/TiffTag$72.class
	bin/pixy/image/tiff/TiffTag$73.class
	bin/pixy/image/tiff/TiffTag$74.class
	bin/pixy/image/tiff/TiffTag$75.class
	bin/pixy/image/tiff/TiffTag$76.class
	bin/pixy/image/tiff/TiffTag$77.class
	bin/pixy/image/tiff/TiffTag$78.class
	bin/pixy/image/tiff/TiffTag$79.class
	bin/pixy/image/tiff/TiffTag$8.class
	bin/pixy/image/tiff/TiffTag$80.class
	bin/pixy/image/tiff/TiffTag$81.class
	bin/pixy/image/tiff/TiffTag$9.class
	bin/pixy/image/tiff/TiffTag$Attribute.class
	bin/pixy/image/tiff/TiffTag.class
	bin/pixy/image/tiff/UndefinedField.class
	bin/pixy/io/EndianAwareInputStream.class
	bin/pixy/io/EndianAwareOutputStream.class
	bin/pixy/io/FileCacheRandomAccessInputStream.class
	bin/pixy/io/FileCacheRandomAccessOutputStream.class
	bin/pixy/io/FileCacheSeekableStream.class
	bin/pixy/io/ForwardSeekableStream.class
	bin/pixy/io/IOUtils.class
	bin/pixy/io/MemoryCacheRandomAccessInputStream.class
	bin/pixy/io/MemoryCacheRandomAccessOutputStream.class
	bin/pixy/io/MemoryCacheSeekableStream.class
	bin/pixy/io/PeekHeadInputStream.class
	bin/pixy/io/PropertyUtil.class
	bin/pixy/io/RandomAccessInputStream.class
	bin/pixy/io/RandomAccessOutputStream.class
	bin/pixy/io/ReadStrategy.class
	bin/pixy/io/ReadStrategyII.class
	bin/pixy/io/ReadStrategyMM.class
	bin/pixy/io/SeekableStream.class
	bin/pixy/io/WriteStrategy.class
	bin/pixy/io/WriteStrategyII.class
	bin/pixy/io/WriteStrategyMM.class
	bin/pixy/meta/Metadata.class
	bin/pixy/meta/MetadataReader.class
	bin/pixy/meta/MetadataType.class
	bin/pixy/meta/Thumbnail.class
	bin/pixy/meta/adobe/BlendModeKey.class
	bin/pixy/meta/adobe/Channel.class
	bin/pixy/meta/adobe/ColorSpaceID.class
	bin/pixy/meta/adobe/DDB.class
	bin/pixy/meta/adobe/DDBEntry.class
	bin/pixy/meta/adobe/DataBlockType.class
	bin/pixy/meta/adobe/FilterMask.class
	bin/pixy/meta/adobe/IPTC_NAA.class
	bin/pixy/meta/adobe/IRB.class
	bin/pixy/meta/adobe/IRBThumbnail.class
	bin/pixy/meta/adobe/ImageResourceID.class
	bin/pixy/meta/adobe/JPEGQuality$Format.class
	bin/pixy/meta/adobe/JPEGQuality$ProgressiveScans.class
	bin/pixy/meta/adobe/JPEGQuality$Quality.class
	bin/pixy/meta/adobe/JPEGQuality.class
	bin/pixy/meta/adobe/LayerData.class
	bin/pixy/meta/adobe/Slice$Descriptor.class
	bin/pixy/meta/adobe/Slice.class
	bin/pixy/meta/adobe/Slices$SliceHeader.class
	bin/pixy/meta/adobe/Slices.class
	bin/pixy/meta/adobe/ThumbnailResource.class
	bin/pixy/meta/adobe/UserMask.class
	bin/pixy/meta/adobe/VersionInfo.class
	bin/pixy/meta/adobe/_8BIM.class
	bin/pixy/meta/bmp/BMPMeta$DataTransferObject.class
	bin/pixy/meta/bmp/BMPMeta.class
	bin/pixy/meta/exif/Exif.class
	bin/pixy/meta/exif/ExifTag$1.class
	bin/pixy/meta/exif/ExifTag$2.class
	bin/pixy/meta/exif/ExifTag$3.class
	bin/pixy/meta/exif/ExifTag$4.class
	bin/pixy/meta/exif/ExifTag$5.class
	bin/pixy/meta/exif/ExifTag$6.class
	bin/pixy/meta/exif/ExifTag$7.class
	bin/pixy/meta/exif/ExifTag.class
	bin/pixy/meta/exif/ExifThumbnail.class
	bin/pixy/meta/exif/GPSTag$1.class
	bin/pixy/meta/exif/GPSTag$2.class
	bin/pixy/meta/exif/GPSTag$3.class
	bin/pixy/meta/exif/GPSTag$4.class
	bin/pixy/meta/exif/GPSTag$5.class
	bin/pixy/meta/exif/GPSTag$6.class
	bin/pixy/meta/exif/GPSTag$7.class
	bin/pixy/meta/exif/GPSTag$8.class
	bin/pixy/meta/exif/GPSTag$9.class
	bin/pixy/meta/exif/GPSTag.class
	bin/pixy/meta/exif/InteropTag$1.class
	bin/pixy/meta/exif/InteropTag$2.class
	bin/pixy/meta/exif/InteropTag$3.class
	bin/pixy/meta/exif/InteropTag$4.class
	bin/pixy/meta/exif/InteropTag$5.class
	bin/pixy/meta/exif/InteropTag.class
	bin/pixy/meta/exif/JpegExif.class
	bin/pixy/meta/exif/TiffExif.class
	bin/pixy/meta/gif/GIFMeta$DataTransferObject.class
	bin/pixy/meta/gif/GIFMeta.class
	bin/pixy/meta/icc/ICCProfile$ICCProfileHeader.class
	bin/pixy/meta/icc/ICCProfile.class
	bin/pixy/meta/icc/ProfileTag$TagType.class
	bin/pixy/meta/icc/ProfileTag.class
	bin/pixy/meta/icc/ProfileTagTable$TagEntry.class
	bin/pixy/meta/icc/ProfileTagTable.class
	bin/pixy/meta/image/Comments.class
	bin/pixy/meta/image/ImageMetadata.class
	bin/pixy/meta/iptc/IPTC.class
	bin/pixy/meta/iptc/IPTCApplicationTag$1.class
	bin/pixy/meta/iptc/IPTCApplicationTag$10.class
	bin/pixy/meta/iptc/IPTCApplicationTag$11.class
	bin/pixy/meta/iptc/IPTCApplicationTag$12.class
	bin/pixy/meta/iptc/IPTCApplicationTag$13.class
	bin/pixy/meta/iptc/IPTCApplicationTag$14.class
	bin/pixy/meta/iptc/IPTCApplicationTag$15.class
	bin/pixy/meta/iptc/IPTCApplicationTag$16.class
	bin/pixy/meta/iptc/IPTCApplicationTag$17.class
	bin/pixy/meta/iptc/IPTCApplicationTag$18.class
	bin/pixy/meta/iptc/IPTCApplicationTag$2.class
	bin/pixy/meta/iptc/IPTCApplicationTag$3.class
	bin/pixy/meta/iptc/IPTCApplicationTag$4.class
	bin/pixy/meta/iptc/IPTCApplicationTag$5.class
	bin/pixy/meta/iptc/IPTCApplicationTag$6.class
	bin/pixy/meta/iptc/IPTCApplicationTag$7.class
	bin/pixy/meta/iptc/IPTCApplicationTag$8.class
	bin/pixy/meta/iptc/IPTCApplicationTag$9.class
	bin/pixy/meta/iptc/IPTCApplicationTag.class
	bin/pixy/meta/iptc/IPTCDataSet.class
	bin/pixy/meta/iptc/IPTCEnvelopeTag$1.class
	bin/pixy/meta/iptc/IPTCEnvelopeTag$2.class
	bin/pixy/meta/iptc/IPTCEnvelopeTag.class
	bin/pixy/meta/iptc/IPTCFotoStationTag.class
	bin/pixy/meta/iptc/IPTCNewsPhotoTag.class
	bin/pixy/meta/iptc/IPTCObjectDataTag.class
	bin/pixy/meta/iptc/IPTCPostObjectDataTag.class
	bin/pixy/meta/iptc/IPTCPreObjectDataTag.class
	bin/pixy/meta/iptc/IPTCRecord.class
	bin/pixy/meta/iptc/IPTCTag.class
	bin/pixy/meta/jpeg/AdobeSegment.class
	bin/pixy/meta/jpeg/DuckyDataSet.class
	bin/pixy/meta/jpeg/DuckySegment.class
	bin/pixy/meta/jpeg/DuckyTag.class
	bin/pixy/meta/jpeg/JFIFSegment.class
	bin/pixy/meta/jpeg/JFIFThumbnail.class
	bin/pixy/meta/jpeg/JPEGMeta.class
	bin/pixy/meta/png/PNGMeta.class
	bin/pixy/meta/png/TIMEChunk.class
	bin/pixy/meta/png/TextualChunks.class
	bin/pixy/meta/tiff/TIFFMeta.class
	bin/pixy/string/Base64.class
	bin/pixy/string/StringUtils$1.class
	bin/pixy/string/StringUtils.class
	bin/pixy/string/XMLUtils.class
	bin/pixy/test/TestPixyMetaAndroid.class
	bin/pixy/util/ArrayUtils$1.class
	bin/pixy/util/ArrayUtils.class
	bin/pixy/util/Builder.class
	bin/pixy/util/CollectionUtils.class
	bin/pixy/util/FileUtils$1.class
	bin/pixy/util/FileUtils.class
	bin/pixy/util/IntHashtable$HashEntry.class
	bin/pixy/util/IntHashtable.class
	bin/pixy/util/LangUtils.class
	bin/pixy/util/MetadataUtils.class
	bin/pixy/util/Reader.class
	bin/pixy/util/zip/CRC32.class
	bin/pixy/util/zip/Checksum.class
	build/classes/changelog.txt
	build/classes/pixy/meta/Metadata.class
	build/classes/pixy/meta/adobe/ThumbnailResource.class
	build/classes/pixy/meta/gif/GIFMeta.class
	build/classes/pixy/meta/jpeg/JPEGMeta.class
	build/classes/pixy/meta/png/PNGMeta.class
	build/classes/pixy/meta/tiff/TIFFMeta.class
	build/classes/pixy/test/TestPixyMetaAndroid.class
	build/classes/pixy/util/ArrayUtils.class
	build/doc/allclasses-frame.html
	build/doc/allclasses-noframe.html
	build/doc/index-all.html
	build/doc/overview-frame.html
	build/doc/overview-summary.html
	build/doc/overview-tree.html
	build/doc/package-list
	build/doc/pixy/image/tiff/package-tree.html
	build/doc/pixy/meta/Metadata.html
	build/doc/pixy/meta/MetadataReader.html
	build/doc/pixy/meta/adobe/DDB.html
	build/doc/pixy/meta/adobe/IRB.html
	build/doc/pixy/meta/adobe/ThumbnailResource.html
	build/doc/pixy/meta/adobe/VersionInfo.html
	build/doc/pixy/meta/adobe/package-frame.html
	build/doc/pixy/meta/adobe/package-summary.html
	build/doc/pixy/meta/adobe/package-tree.html
	build/doc/pixy/meta/exif/Exif.html
	build/doc/pixy/meta/exif/JpegExif.html
	build/doc/pixy/meta/exif/TiffExif.html
	build/doc/pixy/meta/exif/package-tree.html
	build/doc/pixy/meta/gif/GIFMeta.html
	build/doc/pixy/meta/gif/package-frame.html
	build/doc/pixy/meta/gif/package-summary.html
	build/doc/pixy/meta/gif/package-tree.html
	build/doc/pixy/meta/icc/ICCProfile.html
	build/doc/pixy/meta/icc/package-tree.html
	build/doc/pixy/meta/image/Comments.html
	build/doc/pixy/meta/image/ImageMetadata.html
	build/doc/pixy/meta/iptc/IPTC.html
	build/doc/pixy/meta/iptc/package-tree.html
	build/doc/pixy/meta/jpeg/AdobeSegment.html
	build/doc/pixy/meta/jpeg/DuckySegment.html
	build/doc/pixy/meta/jpeg/JFIFSegment.html
	build/doc/pixy/meta/jpeg/JPEGMeta.html
	build/doc/pixy/meta/jpeg/package-frame.html
	build/doc/pixy/meta/jpeg/package-summary.html
	build/doc/pixy/meta/jpeg/package-tree.html
	build/doc/pixy/meta/png/PNGMeta.html
	build/doc/pixy/meta/png/TIMEChunk.html
	build/doc/pixy/meta/png/TextualChunks.html
	build/doc/pixy/meta/png/package-frame.html
	build/doc/pixy/meta/png/package-summary.html
	build/doc/pixy/meta/png/package-tree.html
	build/doc/pixy/meta/tiff/TIFFMeta.html
	build/doc/pixy/meta/tiff/package-frame.html
	build/doc/pixy/meta/tiff/package-summary.html
	build/doc/pixy/meta/tiff/package-tree.html
	build/doc/pixy/meta/xmp/XMP.html
	build/doc/pixy/string/package-summary.html
	build/doc/pixy/string/package-tree.html
	build/doc/pixy/util/ArrayUtils.html
	build/doc/pixy/util/Reader.html
	dist/pixymeta-android-1.0-SNAPSHOT.jar
	dist/pixymeta-android.jar
	lib/pixymeta-android-1.0-SNAPSHOT.jar
	lib/pixymeta-android.jar
	pixymeta-android-lib/src/main/java/pixy/meta/adobe/XMP.java
	src/changelog.txt
	src/pixy/meta/adobe/XMP.java
	src/pixy/meta/xmp/XMP.java

k3b committed:
updated gitignore

k3b committed:
fixed merge conflict

k3b committed:
trigger travis rebuild

k3b committed:
Refactored all android dependencies to one single class BitmapNative

k3b committed:
refactored out android independant pixymeta-lib

k3b committed:
fixed build

k3b committed:
made pixymeta-lib also j2se compatible by implementing awt specific pixymeta-j2se-lib

k3b committed:
added j2se demo app with testdata

k3b committed:
made failing demos more robust

k3b committed:
tried different imageio driver

k3b committed:
fixed https://github.com/dragon66/pixymeta/issues/3 png comment failed

k3b committed:
Update README.md

k3b committed:
added api IBitmap.getImage()

dragon66 commented:
@k3b I haven't done github pull and merge before. Since your version has removed all the Eclipse and Maven related build artifacts, will the users still be able to use Maven repository for pixymeta-android?

k3b commented:
In my opinion the intermediate file (*.class) are not neccessary and should not be part of git.
- Eclipse specific files ( .classpath, .project ) and 
- ant build.xml and 
- maven  pom.xml

Sorry my fault: I deleted them because after changing the directory layout they will not work any more.

They should be backed up before the merge and then added again and fixed for the modified directory layout.

according to http://stackoverflow.com/questions/18955149/ignore-classpath-and-project-from-git 
.classpath and .project should not be kept local but not part of git 

```
  > will the users still be able to use Maven repository for pixymeta-android?
```

the gradle script currently has no code to generate maven-pom-files or upload a new release to mavencentral/jcenter. i have no experience with this.

can i put the burden on you to restore fix the deleted files.

k3b commented:
now i have learned how to create the upload part with gradle. if you want i can add the upload code to pixy meta gradle

dragon66 commented:
Go ahead please. Thanks!

k3b committed:
added gradle maven-repository-upload support

k3b committed:
added gradle maven-repository-upload support

k3b committed:
merge from dragon66

k3b committed:
fixed gradle upload and travis build

k3b committed:
gradle use logger.info instead of print so output can be filtered

k3b commented:
I updated the gradle scripts to allow upload to maven repository (at  `oss.sonatype.org` or to a local file directory) using the gradle task `uploadArchives`

for this to work you have to create the file `gradle.properties`. you find a template with comments at `gradle.properties.example`

k3b committed:
fixed maven-upload internal checks

k3b commented:
I fixed the upload script again. now i succeded to upload a snapshot and a non-snapshot to sonatype provided that gradle.properties have the correct settings.

I added the username, password, email-info to the user-global (non-project specific) location at
- C:\Users<<MyUsername>>.gradle\gradle.properties so no more risk to accidentally checkin private data
